### ISSUE TITLE
Better updater

### ISSUE THREAD

**Issue by HarelM** [sqlitebrowser/sqlitebrowser#983]
Would be great if there was a proper updater instead of the popup that tells me to go and download the latest version when opening the app.
Keep up the good work!

**Comment by chrisjlocke** [comment]
This is sort of intentional.  Some people dont want their software updating automatically or to a different version than the one they have.  A message popping up seemed a happy compromise.

**Comment by HarelM** [comment]
I'm not arguing about the necessity of the pop-up :-)
Just the trip to the site, download and install which can be done as part of the application.
See Notepad++ for example, and many others.
If a user doesn't want to update he should just click cancel.
For me, the necessity to download something and then install it is what sometimes prevents me from updating an application (just sharing my point of view as a user :-))

**Comment by Naihonn** [comment]
I agree with HarelM. It is always nice when you can upgrade so seamlessly. :0) And Notepad++ is indeed quite nice example of that. Although it is downloading installer first but it discards it after that I hope.

**Comment by justinclift** [comment]
Thinking about this more, we could probably add more info to the pop up, and perhaps have a direct "Download" button which fetches it and automatically starts it.

Thoughts?

**Comment by HarelM** [comment]
I would probably give the button a better name such as "Download and install" or "Update now" - download seems to me like it will only download the file without installing it.
Also it's probably worth considering a service that does that already - e.g. chocolatery.

**Comment by justinclift** [comment]
@HarelM Good point about the button name.

With using a service that does it already... yeah.  It just needs someone interested in that side of things to make it happen.  eg write the code the bits and do whatever else is needed (integration? no real idea yet).

On OSX there's "Sparkle" which has a decent reputation, and we've had a few people over the years say they'll pick up the task.  But it's always fallen though.  From (possibly dodgy) memory, due to the task being bigger than first guessed. :wink:

If it's something you'd be interested in making happen though, then by all means go for it. :smile:
